src/engine/strat_engine/backstore/util.rs

```diff
@@ -36,3 +39,44 @@ pub fn hw_lookup(dev_node_search: &Path) -> EngineResult<Option<String>> {
 
     result
 }
+
+// Probes a block device using the libblkid library and returns Some(None) if the device is empty or
+// Some(String) with the String having similar content as the output of `blkid -p <device>.
+pub fn device_signature(path: &Path) -> Result<Option<String>, blkid::BlkidError> {
+
+    let d = blkid::BlkId::new(path)?;
+
+    d.enable_partitions()?
+        .enable_superblocks()?
+        .set_superblock_flags(blkid_sys::BLKID_SUBLKS_USAGE | blkid_sys::BLKID_SUBLKS_TYPE |
+                              blkid_sys::BLKID_SUBLKS_BADCSUM |
+                              blkid_sys::BLKID_SUBLKS_UUID)?;
+
+    // There is potential for the probe to fail if the device is in use/locked, lets loop a bit
+    // to ride out these potentially temporary errors.
+    let mut num_retries = 5;
+    let probe_result = loop {
+        match d.do_safe_probe() {
+            Ok(p) => break p,
+            Err(e) => {
+                num_retries -= 1;
+                if num_retries == 0 {
+                    return Err(e);
+                }
+                thread::sleep(time::Duration::from_millis(100));
+            }
+        }
+    };
+
+    Ok(match probe_result {
+           1 => None,
+           -2 => Some(String::from("Ambiguous device usage detected")),
```

Comment:
Why not just return an error when there is this ambiguity and return the map that ```get_values_map()``` returns directly, rather than converting it down to a bunch of strings joined with spaces?

Reply:
My intention was that this function would return some kind of useful string which would be returned up to user space, thus the documentation that states that it's similar to `blkid` output.  If I return the map then the caller would likely need to do this.  However, as Andy suggested we will need to return the map if we are going to leverage blkid to return information needed by Stratis instead of Stratis parsing it's own BDA.